test/semian_test.rb

```diff
@@ -15,7 +15,7 @@ def test_unsupported_acquire_yields
   end
 
   def test_register_with_circuit_breaker_missing_options
-    assert_raises ArgumentError, "CI stub" do
+    assert_raises ArgumentError do
```

Comment:
Testing the exception message isn't straight forward and the only way I've had it work in the past is with two assertions. One for the `assert_raises` and the part `assert_equals`/`assert_match`. In your case:

```rb
def test_register_with_circuit_breaker_missing_options
  exception = assert_raises ArgumentError do
    Semian.register(
      :testing,
      error_threshold: 2,
      error_timeout: 5,
      bulkhead: false,
    )
  end
  assert_equal("CI stub", exception.message)
end
```

(I haven't used this in quite a while so YMMV)

I did manage to find others hitting this scenario in the minitest GitHub issue queue and one of the solutions was to add an assertion like `assert_raises_with_message` (see seattlerb/minitest#627) but it's been stagnating without any progress for a while so I wouldn't be holding my breathe on it.

Reply:
Do you mind if I make a different exception type like `MissingOptionsError`?  I would need to decide whether it should inherit from `Semian::BaseError` or `ArgumentError` for compatibility.

Reply:
@jacobbednarz I realized changing the exception type may require a major version bump, which I'd like to avoid.  For that reason I did go with your approach.  Let me know if I can :shipit:.

Reply:
`ArgumentError` makes sense to me here. You're expecting a particular set of arguments and if that isn't met, you are raising an exception.